x-pack/plugins/observability_solution/infra/public/apps/logs_app.tsx

```diff
@@ -57,7 +58,26 @@ const LogsApp: React.FC<{
   storage: Storage;
   theme$: AppMountParameters['theme$'];
 }> = ({ core, history, pluginStart, plugins, setHeaderActionMenu, storage, theme$ }) => {
-  const uiCapabilities = core.application.capabilities;
+  const { logs, discover, fleet } = core.application.capabilities;
+
+  const LogsRouteComponent =
+    discover?.show && fleet?.read ? (
+      <Route
+        path="/"
+        exact
+        render={() => {
+          plugins.share.url.locators
+            .get<AllDatasetsLocatorParams>(ALL_DATASETS_LOCATOR_ID)
+            ?.navigate({});
+
+          return null;
+        }}
+      />
+    ) : (
+      // This needs to redirect to /categories once stream is deprecated when the below ticket is implemented
+      // https://github.com/elastic/kibana/issues/193321
+      <RedirectWithQueryParams from={'/'} to={'/stream'} exact />
```

Comment:
> [!TIP]
> There is not need for this redirect, the above route was introduced as a workaround to redirect from this app routing to Logs Explorer. If Logs Explorer is not accessible, you can simply conditionally render this route, and the router will match the next route at line 89.
```ts
<Routes>
  {Boolean(discover?.show && fleet?.read) && (
    <Route
      path="/"
      exact
      render={() => {
        plugins.share.url.locators
          .get<AllDatasetsLocatorParams>(ALL_DATASETS_LOCATOR_ID)
          ?.navigate({});

        return null;
      }}
    />
  )}
  <Route path="/link-to" component={LinkToLogsPage} />
  {uiCapabilities?.logs?.show && <Route path="/" component={LogsPage} />}
</Routes>
```

Reply:
This will also require to handle the fallback chain once the main app is reached, which is something I'm working on https://github.com/elastic/kibana/pull/194519

Reply:
In this case, if you want to make it work, you need to restore the change I made here sometime ago: https://github.com/tonyghiani/kibana/commit/72284dbc1a5e1729305914b13b16fb2322965945#diff-2e08d3a5f5a2bf218ac914659e6ee909c4b8bdcc10a9b6ba14efd7834834e985L110

I'll update the conflicts then once I get my PR ready to merge 👌

Reply:
No need to conditionally render it, if logs explorer is enabled it doesn't get there down in the tree, so you can leave it. I'll have to restore it anyway based on the existence of Logs Stream, it should work fine